airflow/providers/qubole/operators/qubole.py

```diff
@@ -217,7 +217,7 @@ class QuboleOperator(BaseOperator):
         'cluster_label',
     )
 
-    template_ext: Iterable[str] = ('.txt',)
+    template_ext: Sequence[str] = ('.txt',)
```

Comment:
Maybe this should have been `Collection` instead of `Sequence`? Or does anything in Airflow requires these to be deterministically ordered?

Reply:
This field is defined in the base class as `Sequence`. It used to be `Iterable` and was changed [here](https://github.com/apache/airflow/commit/59e4b78daa3496cb0358ce34aeb5ebf6f5565ce0#diff-848f325ace55b3504e8052fecdb53c0f295c891b67a6d90e9341cbe79cc545fbR428). I changed it to match base class and not sure if order here metters at all. BTW, it was suggested by you :) https://github.com/apache/airflow/pull/20034#discussion_r762426218

Reply:
Yep. I already implemented Sequence for ALL operators in #20571 :).

Reply:
And #20608

Reply:
So better not change it again - unless we have a good reson to.

Reply:
The reason being the `tuple` change below indicating `Sequence` (which `set` does not implement) is potentially the wrong protocol.

Reply:
The reason for below set-tuple conversion is to remove any duplicates and hence `set` with `union` operator was used prior to conversion. It's similar to one in here:
https://github.com/apache/airflow/blob/84a4bcca293ac1c85a91dfc7f97938ff2fd9c564/airflow/providers/qubole/operators/qubole_check.py#L108-L110
Well, it also means that if we dont need dups then Set type would be more appropriate

Reply:
If we choose another protocol - fine with me. That''s yet andother 280 files change :) . 

Anyone happy to do it ?

Reply:
I can change them to `Set` type if we are in an agreement?

Reply:
As a caution I will also need to change the type of values it (template_fields) gets assigned to, which are mostly `tuple` and will become all `set`. Also documentation needs to be amended as there were no much emphasis on the type of these fields and list also were used: https://airflow.apache.org/docs/apache-airflow/stable/concepts/operators.html#jinja-templating
This means we also need to notify users about this change. 
Alternatively, we can leave them as is and let users to use list, tuple etc. to provide simplicity of use and internally carry on removing duplicates as they (dups) dont make any sense. WDYT?

Reply:
I’d say let’s keep `Sequence` and do the mass-change later. That makes the commit history easier to reason with.